### PR TITLE
For of

### PR DESCRIPTION
**Pull request by aminya**
Optimizes some of the `for-of` loops.

### CODE REVIEW

**Comment by aminya** [comment]
Measuring the performance is hard in such a project that is tied to the Atom, but we get some improvements here and there:
Before: 

After:

File: lib/storage.js [hunk 1]
```diff
@@ -87,8 +87,9 @@ class Storage{
 	 * @return {Boolean}
 	 */
 	isProjectRelated(path){
-		for(const root of atom.project.rootDirectories){
-			const projectPath = root.path;
+		const rootDirectories = atom.project.rootDirectories
```
Comment (Alhadis) [inline lib/storage.js]: Destructuring should be used, and a semicolon added:

~~~js
const {rootDirectories} = atom.project;
~~~

File: lib/ui.js [hunk 2]
```diff
@@ -120,8 +120,11 @@ class UI {
 	
 	
 	observeFiles(fn){
-		for(const editor of atom.textEditors.editors)
+		const editors = atom.textEditors.editors
```
Comment (Alhadis) [inline lib/ui.js]: Same comment as above:

~~~js
const {editors} = atom.textEditors;
~~~

File: lib/icons/icon-tables.js [hunk 3]
```diff
@@ -69,11 +69,13 @@ class IconTables{
 			: [cache.fileName,      this.fileIcons.byName];
 		const cached = cachedIcons.get(name);
 		if(cached) return cached;
-		for(const icon of icons)
+		for (let i = 0, l = icons.length; i < l; ++i) {
```
Comment (Alhadis) [inline lib/icons/icon-tables.js]: Please remove the bracket spacing from each loop:

~~~diff
-for (let i = 0, l = icons.length; i < l; ++i) {
+for(let i = 0, l = icons.length; i < l; ++i){
~~~

**Comment by aminya** [comment]
Your eslint plugin is broken. I fixed some of your rules manually.

**Comment by Alhadis** [comment]
> Your Eslint plugin is broken

Could you be more specific? What exactly was the error?

> I fixed some of your rules manually.

I don't know what you changed, but the PR is now littered with diff-noise. [Take a look](https://github.com/file-icons/atom/pull/805/files).

**Comment by aminya** [comment]
> 
> 
> > Your Eslint plugin is broken
> 
> Could you be more specific? What exactly was the error?
> 

Can't copy
image

**Comment by aminya** [comment]
> 
> I don't know what you changed, but the PR is now littered with diff-noise. [Take a look](https://github.com/file-icons/atom/pull/805/files).

These are trailing spaces (usually hidden) that are removed.
image

**Comment by Alhadis** [comment]
First, the ESLint error is being caused by a reference to `@typescript-eslint/eslint-plugin`, which this package doesn't even use. Going by the screenshot, it's being referenced from `..\.eslintrc`, which I assume is located in the directory containing your fork.

> These are trailing spaces (usually hidden) that are removed.

Yes, I know. I'd appreciate it if you could revert that.

**Comment by aminya** [comment]
> 
> 
> First, the ESLint error is being caused by a reference to `@typescript-eslint/eslint-plugin`, which this package doesn't even use. Going by the screenshot, it's being referenced from `..\.eslintrc`, which I assume is located in the directory containing your fork.
> 

That was strange. It was in another directory, but was being picked by the eslint-plugin

**Comment by aminya** [comment]
I reverted the trailing space issue because of diff noise. Fix the CI yourself.

**Comment by Alhadis** [comment]
> That was strange. It was in another directory, but was being picked by the eslint-plugin

Yeah, that's long been a source of confusion for ESLint users. That's why personal config files are [getting deprecated](https://eslint.org/docs/user-guide/configuring#personal-configuration-file-deprecated) in the next major version.

**Comment by Alhadis** [comment]
Thanks. Not sure what's causing the latest CI error, but it's [unrelated](https://travis-ci.org/github/file-icons/atom/jobs/668026389#L753-L768) to this PR.

**Comment by aminya** [comment]
You're welcome.

I left some other loops that I haven't benchmarked yet (like `for ([a,b] of c)`).

**Comment by Alhadis** [comment]
Silly me, I neglected to check that each of the iterables in question was really an `Array`. The `LinguistStrategy` loop you modified had been iterating through a `Set`, hence the broken tests. I've reverted that particular change.

**Pull request merged**